### ISSUE TITLE
push notification problem

### ISSUE THREAD

**Issue by monikaaggarwal** [Endare/phonegap-push-wp8#1]
How to call push plugin from js?? can you explain me js implementation

**Comment by SamVerschueren** [comment]
Hi, sorry for this not being documented. I will add this in the documentation. 

You only have to add the plugin to the project and it will cal your server with the device I'd. Use this device I'd to send a push notification via mpns (Microsoft push notification service). When I wrote this library it was impossible to catch the data that was send in the notification. I should have a look if this is still the case in phonegap 3.5.

**Comment by monikaaggarwal** [comment]
ya..Please provide me the full documentation to integrate push plugin .I am able to install in windows but don't know how to call these from JavaScript..Please guide me that would be really helpful.

Thank you

**Comment by SamVerschueren** [comment]
Just add it with the command "phonegap local add..." and the plugin will call your server automatically when initialized. So you don't have to call any JavaScript code.

**Comment by monikaaggarwal** [comment]
yes that is the problem it is not initialized automatically. It's just installed in application when I run this that OnInit method not called

**Comment by monikaaggarwal** [comment]
Could you please add some screenshots of running the app with windows phone?? I really need help on this .This is the show stopper for me..

**Comment by SamVerschueren** [comment]
I'm not at home right now. Try this:

``` JavaScript
document.addEventListener('deviceready', function() {
        // Handle push notifications
        // Handle resume
        document.addEventListener('resume', function() {
            PushNotification.getIncoming(handlePush);
        }, false)

        PushNotification.getIncoming(handlePush);

        // Ask if the push is enabled, if not, enable it.
        PushNotification.isPushEnabled(function(hasPush) {
            if(!hasPush) {
                PushNotification.enablePush();
            }
        });

        function handlePush(push) {
             // push object is empty on WP8
        }
    }, false);
```

**Comment by monikaaggarwal** [comment]
hii,

M again not getting any response...Please help me to solve that problem..

**Comment by monikaaggarwal** [comment]
Hi Sam,
I have tried your code several times but failed. Please help me to come out from this problem this is the real headache for me. Please do the needful..

Thanks

**Comment by SamVerschueren** [comment]
I guess I already know what the problem is. Can you mail me at sam.verschueren@endare.com. I will explain you why it isn't working and how to fix it.

**Comment by SamVerschueren** [comment]
Although I'm not sure. Will do my best to help but can't give you any guarantee.
